grafana-plugin/src/containers/RotationForm/RotationForm.tsx

```diff
@@ -357,6 +357,8 @@ export const RotationForm = observer((props: RotationFormProps) => {
             baseDate: value,
             addParams: [activePeriod, 'seconds'],
           })
+        : isMaskedByWeekdays
+        ? dayJSAddWithDSTFixed({ baseDate: value, addParams: [24, 'hours'] })
```

Comment:
<img width="419" alt="Screenshot 2024-07-09 at 15 15 14" src="https://github.com/grafana/oncall/assets/20116910/b8196f19-ff75-4277-b56a-26852c091967">

`shift_end` should be `shift_start + 24 hours` for this configuration ⬆️

Reply:
It's a good practice to avoid nested ternaries. To increase readability, I suggest
```
const handleRotationStartChange = (value: dayjs.Dayjs) => {
    setRotationStart(value);
    setShiftStart(value);

    let addParams;
    if (isLimitShiftEnabled) {
      addParams = [activePeriod, 'seconds'];
    } else if (isMaskedByWeekdays) {
      addParams = [24, 'hours'];
    } else {
      addParams = [recurrenceNum, repeatEveryPeriodToUnitName[recurrencePeriod]];
    }
    setShiftEnd(dayJSAddWithDSTFixed({ baseDate: value, addParams }));
  };
  ```

Reply:
good call! addressed in https://github.com/grafana/oncall/pull/4638/commits/dac509b8189a06131e908b8dbf332542738c8d2f